File: Examples/example/transaction/sim/PrinterOutputStream.cs
/*
 * Created on Apr 23, 2006
 *
 */

using System;
using System.Collections.Generic;
using System.IO;

using net.esper.example.transaction;

using org.apache.commons.logging;

namespace net.esper.example.transaction.sim
{
    /** Subclass to output events in your preferred format.
     * 
     * @author Hans Gilde, Thomas Bernhardt
     *
     */
    public class PrinterOutputStream : OutputStream
    {
        private TextWriter os;

        public PrinterOutputStream(Stream os)
        {
            this.os = new StreamWriter(os);
        }

        public PrinterOutputStream(TextWriter os)
        {
            this.os = os;
        }

        public void Output(IList<TxnEventBase> bucket)
        {
            log.Info(".output Start of bucket, " + bucket.Count + " items");
            foreach (TxnEventBase eventBean in bucket)
            {
                os.WriteLine(eventBean.ToString());
            }
            log.Info(".output End of bucket");
        }

        private static readonly Log log = LogFactory.GetLog(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
    }
}

File: trunk/examples/src/main/java/com/espertech/esper/example/transaction/sim/PrinterOutputStream.java
/*
 * Created on Apr 23, 2006
 *
 */
package com.espertech.esper.example.transaction.sim;

import com.espertech.esper.example.transaction.TxnEventBase;

import java.io.IOException;
import java.io.PrintStream;
import java.util.List;
import java.util.Map;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

/** Subclass to output events in your preferred format.
 * 
 * @author Hans Gilde, Thomas Bernhardt
 *
 */
public class PrinterOutputStream implements OutputStream {
    private PrintStream os;
    
    public PrinterOutputStream(PrintStream os) {
        this.os = os;
    }
    
    public void output(List<TxnEventBase> bucket) throws IOException {
        log.info(".output Start of bucket, " + bucket.size() + " items");
        for(TxnEventBase event:bucket) {
            os.println(event.toString());
        }
        log.info(".output End of bucket");
    }
    private static final Log log = LogFactory.getLog(PrinterOutputStream.class);
}
